### PR TITLE
[matplotplusplus] Add new port

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- What does your PR fix?
Add new port: https://github.com/alandefreitas/matplotplusplus

- Which triplets are supported/not supported? Have you updated the CI baseline?
N/A. No

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @myd7349 

Is work still being done for this PR?

**Comment by myd7349** [comment]
> 
> 
> Hi @myd7349
> 
> Is work still being done for this PR?

Hi. @NancyLi1013 Sorry for the long delay. I am still working on it. And I will spare some more time on it in the following days.

File: ports/matplotplusplus/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO alandefreitas/matplotplusplus
+    REF e3e48e25505fd5b6cff4d53b0c33c29dcaf9f88e
+    SHA512 4b925171ec89890a84f4de3cba22f6acb477863f7b9c579b8d52b5f4f227a02d4ee75506fe69c859fe3a54c9e7e090bf82b8614fc3da0a3b55721fa73207429b
+    HEAD_REF master
+)
+
+vcpkg_check_features(
+    FEATURES
+        OUT_FEATURE_OPTIONS FEATURE_OPTIONS
```
Comment (NancyLi1013) [inline ports/matplotplusplus/portfile.cmake:19]:
```suggestion
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
```

File: ports/matplotplusplus/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "matplotplusplus",
+  "version-string": "2021-03-07",
```
Comment (NancyLi1013) [inline ports/matplotplusplus/vcpkg.json:3]:
```suggestion
  "version-date": "2021-03-07",
```
This should be version-date in this case. Please see https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#22-package-versions

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by myd7349** [comment]
Hi! @NancyLi1013 Sorry for the long delay. I think I have finished this port. Please review it again.

File: ports/matplotplusplus/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        opengl BUILD_EXPERIMENTAL_OPENGL_BACKEND
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (NancyLi1013) [inline ports/matplotplusplus/portfile.cmake:25]:
```suggestion
```
Please see the reference here https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake/vcpkg_cmake_configure.md

File: ports/matplotplusplus/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "matplotplusplus",
+  "version-date": "2021-04-11",
+  "description": "A C++ graphics library for data visualization",
+  "homepage": "https://alandefreitas.github.io/matplotplusplus/",
+  "license": "MIT",
+  "supports": "!uwp",
```
Comment (NancyLi1013) [inline ports/matplotplusplus/vcpkg.json:7]:
```suggestion
  "supports": "!(arm | uwp)",
```
Seems it also doesn't support arm.

File: scripts/ci.baseline.txt [hunk 5]
```diff
@@ -899,6 +899,10 @@ mathgl:x64-osx=fail
 mathgl:x64-uwp=fail
 matio:x64-linux=fail
 matio:x64-osx=fail
+matplotplusplus:arm64-windows=fail
+# matplotplusplus depends on CImg, which doesn't support UWP(std::system is unavailable).
+matplotplusplus:arm-uwp=fail
+matplotplusplus:x64-uwp=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:905]:
```suggestion
```
Since we have added supports field, it's unnecessary to add these lines here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 5 above)
Comment (myd7349) [reply scripts/ci.baseline.txt]: @NancyLi1013 Thanks for your kind help.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for adding this port @myd7349. 

Last thing needs to confirm the feature status. Can this build fine?

**Comment by myd7349** [comment]
Hi! @NancyLi1013 For `matplotplusplus`, the `OpenGL` backend is still an experimental feature currently. I have tested it with `x86-windows` locally.

**Pull request merged**